core/kazoo_number_manager-1.0.0/src/converters/knm_converter_regex.erl

```diff
@@ -0,0 +1,112 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Peter Defebvre
+%%%-------------------------------------------------------------------
+-module(knm_converter_regex).
+
+-include("../knm.hrl").
+
+-export([
+    normalize/1
+    ,to_npan/1
+    ,to_1npan/1
+]).
+
+-define(DEFAULT_E164_CONVERTERS, [{<<"^\\+?1?([2-9][0-9]{2}[2-9][0-9]{6})$">>
+                                   ,wh_json:from_list([{<<"prefix">>, <<"+1">>}])
+                                  }
+                                  ,{<<"^011(\\d*)$|^00(\\d*)$">>
+                                    ,wh_json:from_list([{<<"prefix">>, <<"+">>}])
+                                   }
+                                  ,{<<"^[2-9]\\d{7,}$">>
+                                    ,wh_json:from_list([{<<"prefix">>, <<"+">>}])
+                                   }
+                                 ]).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec normalize(ne_binary()) -> ne_binary().
+normalize(Num) ->
+    to_e164(Num).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec to_npan(ne_binary()) -> ne_binary().
+to_npan(Num) ->
+    case re:run(Num, <<"^\\+?1?([2-9][0-9]{2}[2-9][0-9]{6})$">>, [{'capture', [1], 'binary'}]) of
+        'nomatch' -> Num;
+        {'match', [NPAN]} -> NPAN
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec to_1npan(ne_binary()) -> ne_binary().
+to_1npan(Num) ->
+    case re:run(Num, <<"^\\+?1?([2-9][0-9]{2}[2-9][0-9]{6})$">>, [{'capture', [1], 'binary'}]) of
+        'nomatch' -> Num;
+        {'match', [NPAN]} -> <<$1, NPAN/binary>>
+    end.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec to_e164(ne_binary()) -> ne_binary().
+to_e164(<<$+, _/binary>> = N) -> N;
```

Comment:
I don't remember if it will be removed or just a new specific convertor (knm_converter_account.erl)